#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_tries_option.dpatch by Jonas Meurer <jonas@freesources.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix the --tries commandline option to work again

@DPATCH@
--- cryptsetup/lib/setup.c	2007-03-07 17:08:52.000000000 +0100
+++ cryptsetup/lib/setup.c	2007-03-07 17:08:32.000000000 +0100
@@ -641,7 +641,7 @@
 	}
 	if((r = LUKS_open_any_key(options->device, password, passwordLen, &hdr, &mk, backend) < 0)) {
 		set_error("No key available with this passphrase.\n");
-		goto out1;
+		r = -EPERM; goto out1;
 	}
 	
 	options->offset = hdr.payloadOffset;
